tests/integration/utils.py

```diff
@@ -191,25 +234,40 @@ def fetch_deps_and_check_output(
     if test_params.check_output_json:
         output_json = load_json(os.path.join(output_folder, "output.json"))
         expected_output_json = load_json(os.path.join(test_data_dir, test_case, "output.json"))
+
+        if "project_files" in expected_output_json:
+            _set_tmpdir_path(expected_output_json["project_files"], tmpdir)
+
         log.info("Compare output.json files")
         assert output_json == expected_output_json, f"Expected output.json:/n{output_json}"
 
     if test_params.check_deps_checksums:
-        files_checksums = calculate_files_sha256sum_in_dir(os.path.join(output_folder, "deps"))
+        files_checksums = calculate_files_checksums_in_dir(os.path.join(output_folder, "deps"))
         expected_files_checksums = load_json(
             os.path.join(test_data_dir, test_case, "fetch_deps_sha256sums.json")
         )
         log.info("Compare checksums of fetched deps files")
-        assert (
-            files_checksums == expected_files_checksums
-        ), f"Expected files checksusms:/n{files_checksums}"
+        _validate_checksums(files_checksums, expected_files_checksums)
 
     if test_params.check_vendor_checksums:
-        files_checksums = calculate_files_sha256sum_in_dir(os.path.join(source_folder, "vendor"))
+        files_checksums = calculate_files_checksums_in_dir(os.path.join(source_folder, "vendor"))
         expected_files_checksums = load_json(
             os.path.join(test_data_dir, test_case, "vendor_sha256sums.json")
         )
         log.info("Compare checksums of files in source vendor folder")
         assert files_checksums == expected_files_checksums
 
     return output_folder
+
+
+def _set_tmpdir_path(project_files: list[dict[str, str]], path: Path) -> None:
+    for item in project_files:
+        template = string.Template(item.get("abspath", ""))
+        item["abspath"] = template.safe_substitute(test_case_tmpdir=str(path))
+
+
+def _validate_checksums(actual_checksums: dict[str, str], expected_checksums: dict[str, str]):
+    for dep in expected_checksums.keys():
+        # declaring the error message here since black imposes incorrect formatting otherwise
+        error_message = f"Checksum mismatch for {dep}: expected {expected_checksums[dep]}, but got {actual_checksums[dep]}"
+        assert actual_checksums[dep] == expected_checksums[dep], error_message
```

Comment:
This is only a one-way check - compared to the asserting with `==`, we no longer check that all the `actual_checksums` are really supposed to be there

If this is to improve error output, I might have a better solution: #78